### PR TITLE
add mirror && loader && select

### PR DESCRIPTION
**Pull request by pakoval**

### CODE REVIEW

File: src/assets/styles/components/select.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.select {
+  all: unset;
+  border-bottom: 1px solid $yellow;
+  color: $yellow;
+
+  &__option {
+
+  }
```
Comment (silchencko) [inline src/assets/styles/components/select.scss]: empty?

File: src/components/Loader.vue [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<template>
+  <span class="loader"></span>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from "vue-property-decorator";
+
+@Component
+export default class Loader extends Vue {
+  @Prop({ default: false }) isLoad!: boolean;
```
Comment (silchencko) [inline src/components/Loader.vue]: redundant prop

File: src/views/HomeView.vue [hunk 3]
```diff
@@ -1,5 +1,28 @@
 <template>
   <div class="home">
+    <div class="home__select-block">
+      <div class="home__option">
+        <span class="home__subtitle">Language:</span>
+        <Select
+          :selectedOption="sourceLang"
```
Comment (silchencko) [inline src/views/HomeView.vue]: `sourceLang` is a little confusing
for selectedOption it is a string with language code in letters, but for the `name` it is a text `sourceLang`
Could you create an enum maybe for language codes?
Do you really need the `name` prop?

File: src/views/HomeView.vue [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+          :list="languages"
+          name="sourceLang"
+          @change="changeLanguage"
```
Comment (silchencko) [inline src/views/HomeView.vue]: it's possible too
`@change="(args) => changeLanguage(args, 'sourceLang')"`

File: src/views/HomeView.vue [hunk 5]
```diff
@@ -9,14 +32,20 @@
       </div>
 
       <div class="textarea-block">
-        <TextArea :value="outputTextarea" :isReadonly="true" />
+        <Loader v-if="loading" />
+        <TextArea
+          :value="outputTextarea"
+          :isReadonly="true"
```
Comment (silchencko) [inline src/views/HomeView.vue]: isReadonly="'true'" it's not reactive

File: src/views/HomeView.vue [hunk 6]
```diff
@@ -54,13 +88,26 @@ export default class HomeView extends Vue {
...
     height: "18px",
   };
+  iconMirror: IIcon = {
+    name: "exchange",
+    width: "18px",
+    height: "18px",
+  };
+  languages = langs;
+  targetLang = "uk";
+  sourceLang = "en";
```
Comment (silchencko) [inline src/views/HomeView.vue]: I suggest enum, since there are a few languages

File: src/components/Select.vue [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+<script lang="ts">
+import { Vue, Component, Prop } from "vue-property-decorator";
+import { ILanguage } from "@/views/types";
+
+@Component
+export default class Select extends Vue {
+  @Prop({ default: [] }) list!: ILanguage[];
+  @Prop({ required: true }) name!: string;
+  @Prop({ required: true }) selectedOption!: string;
```
Comment (silchencko) [inline src/components/Select.vue]: I'm not sure about the `selectedOption` 
We pass `selectedOption` from a parent to a child as a prop. And on the other hand `Select` component can change the selected option and then pass it to the parent and parent will pass it back into the `Select` component.
I expected to see a `defaultOption`
Maybe `Select` component will have v-model `<Select v-model=.../>` and changing the selected option we can emit a change event to update the `v-model` in the parent component
Let me know if it makes sense for you and if you have doubts that the suggestion will work

**Review by silchencko — COMMENTED** [review]
Looks good! Have some comments

File: src/views/HomeView.vue (hunk 6 above)
Comment (pakoval) [reply src/views/HomeView.vue]: yep I also thinking about it

File: src/assets/styles/components/select.scss (hunk 1 above)
Comment (pakoval) [reply src/assets/styles/components/select.scss]: at this moment yes. It's for a future

**Review by silchencko — APPROVED** [review]

**Pull request merged**